core/src/main/java/io/undertow/Undertow.java

```diff
@@ -216,15 +215,14 @@ public synchronized void start() {
                         if (listener.sslContext != null) {
                             xnioSsl = new UndertowXnioSsl(xnio, OptionMap.create(Options.USE_DIRECT_BUFFERS, true), listener.sslContext);
                         } else {
-                            OptionMap.Builder builder = OptionMap.builder();
-                            builder.addAll(listener.overrideSocketOptions);
```

Comment:
This is the bug, the option map only contains overrideSocketOptions, not socketOptions.